eth/utils/numeric.py

```diff
@@ -99,3 +99,29 @@ def clamp(inclusive_lower_bound: int,
         return inclusive_upper_bound
     else:
         return value
+
+
+def integer_squareroot(value: int) -> int:
+    """
+    Return the largest integer ``x`` such that ``x**2 <= value``.
+    Ref: https://en.wikipedia.org/wiki/Integer_square_root
+    """
+    if not isinstance(value, int) or isinstance(value, bool):
+        raise ValueError(
+            "Value must be an integer: Got: {0}".format(
+                type(value),
+            )
+        )
+    if value < 0:
+        raise ValueError(
+            "Value cannot be negative: Got: {0}".format(
+                value,
+            )
+        )
+
+    x = value
+    y = (x + 1) // 2
+    while y < x:
+        x = y
+        y = (x + value // x) // 2
```

Comment:
Maybe worth profiling, but I think the body of this function could be replaced by `return int(math.log2(value))` which *should* produce the correct result.

Reply:
Oh, interesting.
I'm inclined to open an issue to the backlog and keep the algorithm as a reference of the spec for a while.

Reply:
backlog: #1550